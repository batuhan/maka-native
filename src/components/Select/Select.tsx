import React, { ReactNode } from "react";
import { Picker } from "react-native";
import styled from "styled-components";

interface Props {
  style?: {};
  selectedValue?: string;
  onValueChange: () => void;
  items?: { value: string; label: string }[];
}

export default function Select({
  items = [],
  onValueChange,
  selectedValue,
  style
}: Props) {
  return (
    <Picker
      selectedValue={selectedValue}
      style={style}
      onValueChange={onValueChange}
    >
      {items.map(item => (
        <Picker.Item label={item.label} value={item.value} />
      ))}
    </Picker>
  );
}
