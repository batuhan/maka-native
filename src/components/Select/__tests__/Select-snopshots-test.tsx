import React from "react";
import renderer from "react-test-renderer";

import Select from "../Select";

describe("Select", () => {
  it("renders correctly", () => {
    const onValueChange = jest.fn();
    const items = [
      {
        label: "select-label",
        value: "select-value"
      }
    ];
    const instance = renderer.create(
      <Select style={{}} onValueChange={onValueChange} items={items} />
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
