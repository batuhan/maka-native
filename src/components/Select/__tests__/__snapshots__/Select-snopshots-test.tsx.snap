// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<View
  style={Object {}}
>
  <RCTPicker
    items={
      Array [
        Object {
          "label": "select-label",
          "textColor": undefined,
          "value": "select-value",
        },
      ]
    }
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    selectedIndex={0}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;
